### PR TITLE
chore: turn on no-unsafe-optional-chaining rule again

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### Description Of Changes
Addressed a comment in https://github.com/sendbird/sendbird-uikit-react/pull/526#discussion_r1188223665 to bring back `no-unsafe-optional-chaining` rule again which is useful to have. 

As I mentioned [here](https://sendbird.slack.com/archives/C050Z6DA70A/p1683624307711019),  I found there're several places(actually many) we're doing optional chaining unnecessarily. Some of them are caught by lint(fixed in this PR) but others are still there. We'd better revisit them too.

### CODE REVIEW

File: src/hooks/VoicePlayer/useVoicePlayer.tsx [hunk 1]
```diff
@@ -70,9 +70,9 @@ export const useVoicePlayer = ({
     play: playVoicePlayer,
     pause: pauseVoicePlayer,
     stop: stopVoicePlayer,
-    playbackTime: currentAudioUnit?.playbackTime * 1000,
```
Comment (AhyoungRyu) [inline src/hooks/VoicePlayer/useVoicePlayer.tsx:73]: Based on `currentAudioUnit` type, don't need the optional chaining here.

File: src/modules/Thread/components/ParentMessageInfo/index.tsx [hunk 2]
```diff
@@ -42,8 +42,8 @@ export default function ParentMessageInfo({
     isOnline,
     userMention,
   } = config;
-  const userId = stores?.userStore?.user?.userId;
```
Comment (AhyoungRyu) [inline src/modules/Thread/components/ParentMessageInfo/index.tsx:45]: Seems every objects' type here are not nullable. So removed all `?` but plz tell me if we need to be safe here. Then I'll correct each types as nullable :)

Comment (sravan-s) [reply src/modules/Thread/components/ParentMessageInfo/index.tsx:45]: Removing all `?` is okay ~

File: src/ui/MessageContent/index.tsx [hunk 3]
```diff
@@ -266,7 +266,7 @@ export default function MessageContent({
               message={message as UserMessage | FileMessage}
               userId={userId}
               isByMe={isByMe}
-              isUnavailable={(replyType === 'THREAD' && (channel?.joinedAt * 1000) > message?.parentMessage?.createdAt)}
+              isUnavailable={(replyType === 'THREAD' && (channel.joinedAt * 1000) > (message.parentMessage?.createdAt ?? 0))}
```
Comment (sravan-s) [inline src/ui/MessageContent/index.tsx:269]: `message.parentMessage ` cc @HoonBaek

File: src/utils/index.ts [hunk 4]
```diff
@@ -163,11 +163,11 @@ export const isVoiceMessage = (message: FileMessage): boolean => {
   if (!(message && isFileMessage(message))) {
     return false;
   }
-  const [mimeType, typeParameter] = message.type?.split(';');
+  const [mimeType, typeParameter] = message.type.split(';');
```
Comment (sravan-s) [inline src/utils/index.ts:166]: cc @HoonBaek

**Review by sravan-s — APPROVED** [review]
Wait for @HoonBaek

File: src/ui/MessageContent/index.tsx (hunk 3 above)
Comment (AhyoungRyu) [reply src/ui/MessageContent/index.tsx:269]: @HoonBaek  Its type looked okay without the optional chaining, but let me know if there's any concern :)

File: src/utils/index.ts (hunk 4 above)
Comment (HoonBaek) [reply src/utils/index.ts:166]: This looks fine

File: src/modules/Thread/components/ParentMessageInfo/index.tsx (hunk 2 above)
Comment (HoonBaek) [reply src/modules/Thread/components/ParentMessageInfo/index.tsx:45]: userId could be undefined
so how about
```
const userId = stores.userStore.user?.userId;
```

File: src/ui/MessageContent/index.tsx (hunk 3 above)
Comment (HoonBaek) [reply src/ui/MessageContent/index.tsx:269]: I'm worried about the case when the channel is null, how can we manage the case

**Review by HoonBaek — APPROVED** [review]
I added the Nullable type and operated the case when the channel is empty

**Comment by AhyoungRyu** [comment]
@HoonBaek Thanks!

**Pull request merged**